backend/dataall/cdkproxy/stacks/cdk_pipeline.py

```diff
@@ -0,0 +1,252 @@
+
+    def get_engine(self):
+        envname = os.environ.get("envname", "local")
+        engine = db.get_engine(envname=envname)
+        return engine
+
+    def get_target(self, target_uri) -> models.DataPipeline:
+        engine = self.get_engine()
+        with engine.scoped_session() as session:
+            return Pipeline.get_pipeline_by_uri(session, target_uri)
+
+    def get_pipeline_environments(self, targer_uri) -> models.DataPipelineEnvironment:
+        engine = self.get_engine()
+        with engine.scoped_session() as session:
+            envs = Pipeline.query_pipeline_environments(
+                session, targer_uri
+            )
+        return envs
+
+    def get_pipeline_cicd_environment(
+        self, pipeline: models.DataPipeline
+    ) -> models.Environment:
+        envname = os.environ.get("envname", "local")
+        engine = db.get_engine(envname=envname)
+        with engine.scoped_session() as session:
+            return Environment.get_environment_by_uri(session, pipeline.environmentUri)
+
+    def get_env_team(self, pipeline: models.DataPipeline) -> models.EnvironmentGroup:
+        engine = self.get_engine()
+        with engine.scoped_session() as session:
+            env = Environment.get_environment_group(
+                session, pipeline.SamlGroupName, pipeline.environmentUri
+            )
+        return env
+
+    def get_dataset(self, dataset_uri) -> models.Dataset:
+        engine = self.get_engine()
+        with engine.scoped_session() as session:
+            ds = Dataset.get_dataset_by_uri(
+                session, dataset_uri
+            )
+        return ds
+
+    def __init__(self, scope, id, target_uri: str = None, **kwargs):
+        kwargs.setdefault("tags", {}).update({"utility": "dataall-data-pipeline"})
+        super().__init__(
+            scope,
+            id,
+            env=kwargs.get("env"),
+            stack_name=kwargs.get("stack_name"),
+            tags=kwargs.get("tags"),
+            description="Cloud formation stack of PIPELINE: {}; URI: {}; DESCRIPTION: {}".format(
+                self.get_target(target_uri=target_uri).label,
+                target_uri,
+                self.get_target(target_uri=target_uri).description,
+            )[
+                :1024
+            ],
+        )
+
+        # Configuration
+        self.target_uri = target_uri
+
+        pipeline = self.get_target(target_uri=target_uri)
+        pipeline_environment = self.get_pipeline_cicd_environment(pipeline=pipeline)
+        pipeline_env_team = self.get_env_team(pipeline=pipeline)
+        # Development environments
+        development_environments = self.get_pipeline_environments(targer_uri=target_uri)
+
+        # Create CodeCommit repository and mirror blueprint code
+        code_dir_path = os.path.realpath(
+            os.path.abspath(
+                os.path.join(
+                    __file__, "..", "..", "..", "..", "blueprints", "cdk_data_pipeline_blueprint"
```

Comment:
this seems risky. is there a better way of locating the "blueprints" directory? using this many relative paths could make debugging difficult if project structure changes even a little